### ISSUE TITLE
Don't fail when nomad restarts some jobs (Bad Nomad)

### ISSUE THREAD

**Issue by arielsvn** [AlexsLemonade/refinebio#1487]
### Context

https://github.com/alexslemonade/refinebio/issues/1391#issuecomment-520982741

Noticed that we have a bunch of jobs for which their failure reason is `ProcessorJob has already completed with a fail - why are we here again? Bad Nomad!`

### Problem or idea

@kurtwheeler has the theory that the nomad job gets interrupted and then restarted by nomad. We haven't been able to avoid nomad from restarting jobs, but we can make sure jobs get restarted properly when this happens instead of failing.

At the time of writing, there were 5813 processor jobs with this error.

```sql
data_refinery=> select count(*) from processor_jobs where failure_reason='ProcessorJob has already completed with a fail - why are we here again? Bad Nomad!';
 count 
-------
  5813
(1 row)
```

### Solution or next step

We'll need to modify this part in `start_job`

https://github.com/AlexsLemonade/refinebio/blob/90a0286/workers/data_refinery_workers/processors/utils.py#L170-L197

**Comment by kurtwheeler** [comment]
I do think that we can go ahead and retry those 5.8k jobs that you found since the one you retried worked. Once we get around to fixing this we can rerun any that have failed like this since and then hopefully there won't be any more after that!

**Comment by arielsvn** [comment]
ok I just ran it for the jobs where 

```

**Comment by arielsvn** [comment]
Fixed by #1541. There we also added a command that will allow to retry samples that failed with this error.
